gradio/blocks.py

```diff
@@ -2309,6 +2309,7 @@ def launch(
         node_server_name: str | None = None,
         node_port: int | None = None,
         ssr_mode: bool | None = None,
+        pwa: bool = False,
```

Comment:
What do you think about calling this parameter `manifest` and letting it be either a `bool` or a `str` path to a user-specified `manifest.json` file if the user wants more control over it? 
```suggestion
        manifest: bool | str = False,
```

Reply:
Actually thinking about this some more, maybe we can enable this by default in Spaces, in which case, we could do
```suggestion
        manifest: bool | None | str = None,
```
and if a user doesn't override `manifest`, we automatically set it to `True` if on Spaces and `False` otherwise

Reply:
Thanks!

Will set the Spaces default 👍 

Also, the `manifest` param has extensibility benefits but I also have some following ideas:
1. In Gradio's case, `manifest.json` is used only for PWA, so `pwa` may look clearer, hiding other things that users don't have to care. (This can be true if most Gradio users just want to use the PWA feature but don't want to think about things behind it e.g. manifest.json. wdyt? being simple vs detailed?)
    * If we will need to provide the full `manifest.json` configuration in the future, we can add the `manifest` param at that moment and `pwa` can work as a shorthand for it.
2. When we provide more flexible API to configure the `manifest.json`, defining the schema with TypedDict or Pydantic like this would be helpful rather than just allowing to pass the JSON string (and this may be too much at this moment):
```python
class ManifestJson(TypedDict):
    name: str
    version: str
    # ... and more fields

def launch(
    # ...
    pwa: bool | ManifestJson = False,
):
    # ...
```

Reply:
My suggestion is just to provide `pwa: bool | None` for now, and add `manifest: ManifestJson | None` in the future if we find it's needed. This is an opinion biased for simplicity hiding the details, so Gradio's API design policy and/or users' potential demand may override it, wdyt?

Reply:
Sounds good to me, we can discuss the exact implementation later but fine with keeping the boolean .pwa attribute right now 👍